internal/web_server/web/handlers/api/public_api.go

```diff
@@ -0,0 +1,156 @@
+package api
+
+import (
+	"database/sql"
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/Molnes/Nyhetsjeger/internal/config"
+	"github.com/Molnes/Nyhetsjeger/internal/models/quizzes"
+	"github.com/Molnes/Nyhetsjeger/internal/models/users/user_quiz"
+	"github.com/Molnes/Nyhetsjeger/internal/models/users/user_quiz_summary"
+	utils "github.com/Molnes/Nyhetsjeger/internal/utils"
+	"github.com/Molnes/Nyhetsjeger/internal/web_server/web/views/components/quiz_components/play_quiz_components"
+	"github.com/Molnes/Nyhetsjeger/internal/web_server/web/views/pages/quiz_pages"
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+)
+
+type publicApiHandler struct {
+	sharedData *config.SharedData
+}
+
+// Creates a new PublicApiHandler
+func NewPublicApiHandler(sharedData *config.SharedData) *publicApiHandler {
+	return &publicApiHandler{sharedData}
+}
+
+// Registers the public api handlers to the given echo group
+func (h *publicApiHandler) RegisterPublicApiHandlers(g *echo.Group) {
+	g.POST("/user-answer", h.postAnswer)
+	g.GET("/question", h.getQuestion)
+	g.POST("/generate-summary", h.postGenerateSummary)
+}
+
+// Handles a post request with question answer from a guest user.
+func (h *publicApiHandler) postAnswer(c echo.Context) error {
+	questionID, err := uuid.Parse(c.QueryParam("question-id"))
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "Invalid or missing question-id")
+	}
+	pickedAnswerID, err := uuid.Parse(c.FormValue("answer-id"))
+	if err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "Invalid or missing answer-id in formdata")
```

Comment:
Shouldn't these errors be in Norwegian?

Reply:
this should only error if someone monkeys with the urls and local storage.... i guess i can translate them